src/app/SmartR/components/controllers/correlation.js

```diff
@@ -0,0 +1,54 @@
+//# sourceURL=correlation.js
+
+'use strict';
+
+angular.module('smartRApp').controller('CorrelationController',
+    ['$scope', 'smartRUtils', 'commonWorkflowService', function($scope, smartRUtils, commonWorkflowService) {
+
+        commonWorkflowService.initializeWorkflow('correlation', $scope);
+
+        $scope.fetch = {
+            disabled: false,
+            running: false,
+            loaded: false,
+            conceptBoxes: {
+                datapoints: {concepts: [], valid: false},
+                annotations: {concepts: [], valid: true}
+            }
+        };
+
+        $scope.runAnalysis = {
+            disabled: true,
+            running: false,
+            scriptResults: {},
+            params: {
+                method: 'pearson',
+                transformation: 'raw'
+            }
+        };
+
+        $scope.$watch('runAnalysis.params.transformation', function(newValue, oldValue) {
```

Comment:
why do we do a watch over transformation and switch correlation analysis? wouldn't watching a flag indicating the type of correlation (e.g. var corrType = 'pearson'/'spearman') be clearer?